### PR TITLE
Synchronize

### PR DESCRIPTION
**Pull request by SergeyTsimfer**
Update notebooks to the latest (and greatest) version of seismiQB

### CODE REVIEW

File: models/Horizon extrapolation/Horizon_extension.ipynb [hunk 1]
```diff
@@ -0,0 +1,790 @@
+{
```
Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p><code>in the previous notebooks</code></p><p> to cover -&gt; to fill in</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>model also sees</p><p>or</p><p>a part of target horizon is also fed into NN</p><p>partial -&gt; incomplete</p><p>trained to retrieve missing </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>You don't need some of them</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>derived?</p><p>you can find more info about &lt;&gt; in &lt;&gt;</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>dataset.labels[0]</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p><code>the greener the better</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>, that will be explained</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>use the remaining...</p><p>transposing xline oriented crops</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>what is data?</p><p>what is leave nonzero?</p><p>articles are sad</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>at the seismic data, full mask that is to be our NN target, and the thinned out version of it that is fed to the model </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>Combine this comment with the previous markdown cell; combine the code with the previous code cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>Combine this comment with the previous markdown cell; combine the code with the previous code cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>ylabel is not needed</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>To fill in</p><p>Unclear</p><p>No need to mention exact mechanism of crop generation: just say that they are generated near boundaries</p><p> </p><p>Are there any unimportant steps in the inference pipeline?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>Newlines are a blessing</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p><code>one some than some</code></p><p>As we could see previously, the horizon that we work with contains only a small amount of tiny holes; thus, we can set the number of performed steps to a small number.</p><p>Let's write like this is English language, not Russian</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>You either start all of the comments with a capital letter, or None of them</p><p>There is potential for visual clarity improvements by using techniques like newlines, spaces, comments, etc</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>So, did we?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (SergeyTsimfer) [inline models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>You can actually compare this horizon to the one you began with</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Comment (dmylzenova) [reply models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>I say  "xline oriented crops have a shape (1, 64, 64)" </p>

Comment (dmylzenova) [reply models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>Why?</p>

Comment (dmylzenova) [reply models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>Why?</p>

Comment (dmylzenova) [reply models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>It doesn't matter so much as it does in the previous comment, but anyway, what's the point?</p>

Comment (SergeyTsimfer) [reply models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>Too bad that is incorrect: as default crop dimensionality is <code>inline, crossline, depth</code>,  the inline-oriented crop (along one chosen inline) has the shape of (1, #, #), while xline-oriented one has the shape of (#, 1, #). Anyway, the comment was about the last action, <code>adaptive reshape</code>,  which transposes xline oriented crops, not the iline oriented ones.</p>

Comment (SergeyTsimfer) [reply models/Horizon extrapolation/Horizon_extension.ipynb:1]: <p>The output of the previous cell is the return value of <code>Pipeline.run</code>,  which does not provide any information and only confuses the reader. IMO</p>

Comment (dmylzenova) [reply models/Horizon extrapolation/Horizon_extension.ipynb:1]: I would say the opposite - inline oriented meaning that a slide is parallel to the inline axis.

Comment (dmylzenova) [reply models/Horizon extrapolation/Horizon_extension.ipynb:1]: I know what the comment was about, I argued on the definition of "xline oriented."

File: scripts/geometry_quality_map.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+""" !!. """
```
Comment (dmylzenova) [inline scripts/geometry_quality_map.py:1]: I tried to run this file for cube_2 but it failed with following error:

> ../seismiqb/src/metrics.py:521: RuntimeWarning:
> 
> Mean of empty slice

Comment (SergeyTsimfer) [reply scripts/geometry_quality_map.py:1]: That is a warning, not an error

File: scripts/convert_to_hdf5.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+# Argname, description, dtype, default
+ARGS = [
+    ('cube-path', 'path to the SEG-Y cube to convert to HDF5', str, None),
+]
+
+
+if __name__ == '__main__':
+    config = make_config('Convert SEG-Y cube to a HDF5 one.', ARGS,
```
Comment (roman-kh) [inline scripts/convert_to_hdf5.py:18]:
```suggestion
    config = make_config('Convert SEG-Y cube to HDF5.', ARGS,
```

File: seismiqb/src/controllers/best_practices.py [hunk 4]
```diff
@@ -8,7 +8,7 @@
 class Dice(nn.Module):
```
Comment (roman-kh) [inline seismiqb/src/controllers/best_practices.py:8]: Why is Dice in best_practices???

Comment (SergeyTsimfer) [reply seismiqb/src/controllers/best_practices.py:8]: Because we can't update `seismiQB` to master version of `BatchFlow` (and use the freshly added loss-functions)

File: datasets/01_Convert_cubes.ipynb [hunk 5]
```diff
@@ -0,0 +1,283 @@
+{
```
Comment (dmylzenova) [inline datasets/01_Convert_cubes.ipynb:1]: <p>remove some extra #</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: datasets/03_Cube_metrics.ipynb [hunk 6]
```diff
@@ -0,0 +1,231 @@
+{
```
Comment (dmylzenova) [inline datasets/03_Cube_metrics.ipynb:1]: <p>Can we add or provide a link to&nbsp;a more detailed description of the metrics computed below? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

File: datasets/04_Horizon_metrics.ipynb [hunk 7]
```diff
@@ -0,0 +1,258 @@
+{
```
Comment (dmylzenova) [inline datasets/04_Horizon_metrics.ipynb:1]: <p>Is usage of the same header as in "03_Cube_metrics" intentional?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (dmylzenova) [inline datasets/04_Horizon_metrics.ipynb:1]: <p>Comments in demonstration notebooks look suspicious to me or at least somehow should be commented</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: models/Carcass_interpolation/02_Iterative_research.ipynb [hunk 8]
```diff
@@ -0,0 +1,491 @@
+{
```
Comment (dmylzenova) [inline models/Carcass_interpolation/02_Iterative_research.ipynb:1]: <p>Can you add some comments to this notebook, e.g. don't read it or smth? Or move to researches folder</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

File: models/Carcass_interpolation/02_Iterative_results.ipynb [hunk 9]
```diff
@@ -0,0 +1,728 @@
+{
```
Comment (dmylzenova) [inline models/Carcass_interpolation/02_Iterative_results.ipynb:1]: <p>yes, let's move it to a separate folder</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

File: seismiqb/src/crop_batch.py [hunk 10]
```diff
@@ -347,13 +352,15 @@ def create_masks(self, ix, dst, src='slices', width=3, src_labels='labels', indi
...
+                break
         return mask
 
 
     @action
     @inbatch_parallel(init='indices', post='_post_mask_rebatch', target='for',
-                      src='masks', threshold=0.8, passdown=None)
-    def mask_rebatch(self, ix, src='masks', threshold=0.8, passdown=None):
+                      src='masks', threshold=0.8, passdown=None, axis=-1)
+    def mask_rebatch(self, ix, src='masks', threshold=0.8, passdown=None, axis=-1):
```
Comment (dmylzenova) [inline seismiqb/src/crop_batch.py:363]: add parameters description to the docs

File: seismiqb/src/cubeset.py [hunk 11]
```diff
@@ -515,7 +515,7 @@ def compare_to_labels(self, horizon, src_labels='labels', offset=0, absolute=Tru
                                                                 printer=printer, hist=hist, plot=plot)
 
 
-    def show_slide(self, idx=0, n_line=0, axis='iline', mode='overlap', backend='matplotlib', **kwargs):
+    def show_slide(self, loc, idx=0, axis='iline', zoom_slice=None, mode='overlap', backend='matplotlib', **kwargs):
```
Comment (dmylzenova) [inline seismiqb/src/cubeset.py:518]: update the docs

File: seismiqb/src/horizon.py [hunk 12]
```diff
@@ -1156,8 +1156,10 @@ def evaluate(self, supports=50, plot=True, savepath=None, printer=print, **kwarg
         Number of holes inside borders:                   {self.number_of_holes}
         """
         printer(dedent(msg))
-        return self.horizon_metrics.evaluate('support_corrs', supports=supports, agg='nanmean',
-                                             plot=plot, savepath=savepath, **kwargs)
+        if compute_metric:
+            return self.horizon_metrics.evaluate('support_corrs', supports=supports, agg='nanmean',
+                                                 plot=plot, savepath=savepath, **kwargs)
+        return None
```
Comment (dmylzenova) [inline seismiqb/src/horizon.py:1162]: why?

**Comment by dmylzenova** [comment]
Also, readme must be updated too

File: seismiqb/src/horizon.py (hunk 12 above)
Comment (SergeyTsimfer) [reply seismiqb/src/horizon.py:1162]: Sometimes you just want to print a number of statistics
